crates/sui-proxy/src/ip.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use ipnetwork::IpNetwork;
+use multiaddr::Multiaddr;
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+
+pub fn to_multiaddr(addr: IpAddr) -> Multiaddr {
+    match addr {
+        IpAddr::V4(a) => Multiaddr::from(a),
+        IpAddr::V6(a) => Multiaddr::from(a),
+    }
+}
+
+/// is_private makes a decent guess at determining of an addr is publicly routable.
+pub fn is_private(addr: IpAddr) -> bool {
+    match addr {
+        IpAddr::V4(a) => is_private_v4(a),
+        IpAddr::V6(a) => is_private_v6(a),
+    }
+}
+
+/// is_private_v4 will say just that, is it private? we ignore 169.254.0.0/16 in this consideration
+fn is_private_v4(addr: Ipv4Addr) -> bool {
+    // special case we will allow
+    let allowed_private: IpNetwork = "169.254.0.0/16".parse().unwrap();
```

Comment:
what's the significance of this address?

Reply:
its not routable.  i tend to view 169.254.0.0/16 addresses as special local network cases.  for lack of any better choice, i picked this range since the internal code requires a multiaddr type, which in turn requires an ip address of some kind.  there is plenty of space in a /16 and it's a special case. it seemed to fit the need.